orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/AbstractClusterWideClouddriverTask.groovy

```diff
@@ -201,6 +176,54 @@ abstract class AbstractClusterWideClouddriverTask extends AbstractCloudProviderA
     return filterParentDeploys(stage, account, location, serverGroups)
   }
 
+  static List<TargetServerGroup> parentDeployedServerGroups(Stage stage,
+                                                            String account,
+                                                            Location location,
+                                                            List<TargetServerGroup> clusterServerGroups) {
+    //if we are a synthetic stage child of a deploy, don't operate on what we just deployed
+    final Set<String> deployStageTypes = [
+      DeployStage.PIPELINE_CONFIG_TYPE,
+      CopyLastAsgStage.PIPELINE_CONFIG_TYPE,
+      CloneServerGroupStage.PIPELINE_CONFIG_TYPE,
+      CreateServerGroupStage.PIPELINE_CONFIG_TYPE,
+    ]
+    List<TargetServerGroup> deployedServerGroups = []
+
+    stage.ancestors({ Stage ancestorStage, StageBuilder stageBuilder ->
```

Comment:
This will now do a proper ancestor lookup vs. only looking at the immediate parent.